src/app/shared/device-info.service.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from '@angular/core';
+
+// Set the same as the screen sizes in _variables.scss
+const mobileBreakpoint = 780;
+const tabletBreakpoint = 1000;
+
+interface DeviceSize {
+  isMobile: boolean;
+  isTablet: boolean;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DeviceInfoService {
+  private screenWidth: number;
+
+  constructor() {
+    this.screenWidth = window.innerWidth;
+    window.addEventListener('resize', this.updateScreenWidth.bind(this));
+  }
+
+  updateScreenWidth(): void {
+    this.screenWidth = window.innerWidth;
+  }
+
+  getDeviceSize(): DeviceSize {
+    const isMobile = this.screenWidth <= mobileBreakpoint;
```

Comment:
It will be only one true so instead of returning array and  we can just return one so you do not need to calculate all cases. And may be use like 'MOBILE'|'TABLET' in uppercase so it is like keyword.

Reply:
@lmmrssa Updated to use an enum instead, so the device type can be returned